shub/utils.py

```diff
@@ -131,7 +131,13 @@ def _ext(file_path):
 
 def build_and_deploy_egg(project_id):
     """Builds and deploys the current dir's egg"""
-    run('python setup.py bdist_egg')
+    log("Building egg in: %s" % os.getcwd())
+    try:
+        run('python setup.py bdist_egg')
+    except CalledProcessError:
+        # maybe a C extension or distutils package, forcing bdist_egg
```

Comment:
This decision should be logged:

    log("Couldn't build an egg with vanilla setup.py, trying with setuptools...")

After that, LGTM =)